Sources/HttpPipeline/Response.swift

```diff
@@ -1,7 +1,146 @@
 import Foundation
+import MediaType
+import Prelude
 
 public struct Response {
   public private(set) var status: Status
-  public private(set) var headers: [ResponseHeader]
+  public private(set) var headers: [Header]
   public private(set) var body: Data
+
+  public struct Header {
+    public let name: String
+    public let value: String
+
+    public init(_ name: String, _ value: String) {
+      self.name = name
+      self.value = value
+    }
+
+    public static func allow(_ methods: [Method]) -> Header {
+      return .init("Allow", methods.map(^\.description).joined(separator: ", "))
+    }
+
+    public static func contentLength(_ length: Int) -> Header {
+      return .init("Content-Length", String(length))
+    }
+
+    public static func contentType(_ type: MediaType) -> Header {
+      return .init("Content-Type", type.description)
+    }
+
+    public static func location(_ location: String) -> Header {
+      return .init("Location", location)
+    }
+
+    public static func setCookie(_ name: String, _ value: String, _ options: Set<CookieOption> = [])
```

Comment:
I've removed the need for parameter names here. Autocompletion will give `<# name #>` and `<# value #>` hints in this post-`enum` world, and order seems to retain enough semantics here.